### PR TITLE
[KYUUBI #524] Setup Kyuubi Project Improvement Proposals(KBIP)

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Setup Kyuubi Project Improvement Proposals(KBIP).
And will follow it after community email list established.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: docs/community/improvement-proposals.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+- Uphold the quality of the changes, including verifying whether the changes satisfy the goal of the KBIP and are absent of critical bugs before releasing them
+
+## KBIP Process
+### Proposing an KBIP
+
+Anyone may propose an KBIP, using the document template below. Please only submit an KBIP if you are willing to help, at least with discussion.
+
+If an KBIP is too small or incremental and should have been done through the normal JIRA process, a committer should remove the KBIP label.
+
+
```
Comment (turboFei) [inline docs/community/improvement-proposals.md:42]: blank line

File: docs/community/improvement-proposals.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Kyuubi Project Improvement Proposals (KBIP)
+
+The purpose of an KBIP is to inform and involve the user community in major improvements to the Kyuubi codebase throughout the development process, to increase the likelihood that user needs are met.
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:3]: The purpose of a KBIP is to inform and involve the user community in significant improvements to the Kyuubi codebase throughout the development process to increase user needs.

File: docs/community/improvement-proposals.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+KBIPs should be used for significant user-facing or cross-cutting changes, not small incremental improvements. When in doubt, if a committer thinks a change needs an KBIP, it does.
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:5]: small -> minor an KBIP -> a KBIP

File: docs/community/improvement-proposals.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+## What is a KBIP?
+
+An KBIP is similar to a product requirement document commonly used in product management.
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:9]: A

File: docs/community/improvement-proposals.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+An KBIP:
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:11]: A

File: docs/community/improvement-proposals.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+- Is a ticket labeled “KBIP” proposing a major improvement or change to Kyuubi
+- Follows the template defined below
+- Includes discussions on the ticket about the proposal
+
+## Who?
+
+Any **community member** can help by discussing whether an KBIP is likely to meet their needs, and by proposing KBIPs.
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:19]: .... discussing  `whether a KBIP is likely to meet their needs and propose KBIPs.`

File: docs/community/improvement-proposals.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+**Contributors** can help by discussing whether an KBIP is likely to be technically feasible.
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:21]: A

File: docs/community/improvement-proposals.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+**Committers** can help by discussing whether an KBIP aligns with long-term project goals, and by shepherding KBIPs.
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:23]: a

File: docs/community/improvement-proposals.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+**KBIP Shepherd** is a PMC member who is committed to shepherding the proposed change throughout the entire process. Although the shepherd can delegate or work with other committers in the development process, the shepherd is ultimately responsible for the success or failure of the KBIP. Responsibilities of the shepherd include, but are not limited to:
+
+- Be the advocate for the proposed change
+- Help push forward on design and achieve consensus among key stakeholders
+- Review code changes, making sure the change follows project standards
+- Get feedback from users and iterate on the design & implementation
+- Uphold the quality of the changes, including verifying whether the changes satisfy the goal of the KBIP and are absent of critical bugs before releasing them
+
+## KBIP Process
+### Proposing an KBIP
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:36]: A, maybe replace all An with A latter ;)

File: docs/community/improvement-proposals.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+A KBIP document is a short document with a few questions, inspired by the Heilmeier Catechism:
+
+- Q1. What are you trying to do? Articulate your objectives using absolutely no jargon.
+
+- Q2. What problem is this proposal NOT designed to solve?
+
+- Q3. How is it done today, and what are the limits of current practice?
+
+- Q4. What is new in your approach and why do you think it will be successful?
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:52]: `approach, and`

File: docs/community/improvement-proposals.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+- Appendix A. Proposed API Changes. Optional section defining APIs changes, if any. Backward and forward compatibility must be taken into account.
+
+- Appendix B. Optional Design Sketch: How are the goals going to be accomplished? Give sufficient technical detail to allow a contributor to judge whether it's likely to be feasible. Note that this is not a full design document.
+
+- Appendix C. Optional Rejected Designs: What alternatives were considered? Why were they rejected? If no alternatives have been considered, the problem needs more thought.
+
+### Discussing an KBIP
+
+All discussion of an KBIP should take place in a public forum, preferably the discussion attached to the ticket. Any discussions that happen offline should be made available online for the public via meeting notes summarizing the discussions.
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:70]: All discussions

Comment (yaooqinn) [reply docs/community/improvement-proposals.md:70]: Any discussion

File: docs/community/improvement-proposals.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
+# Kyuubi Project Improvement Proposals (KBIP)
```
Comment (yaooqinn) [inline docs/community/improvement-proposals.md:1]: Add the Kyuubi logo like other new added docs

**Review by yaooqinn — APPROVED** [review]

**Comment by turboFei** [comment]
thanks for your review, merging to master
